Notification boolean

gaemers3232 opened the pull request:
Boolean for notification request added

gaemers3232 committed:
Notification boolean

Boolean for notification request added

kvangork commented on app/src/main/java/csustan/edu/cs4800/warriordiningapp/MainActivity.java:
```diff
@@ -1,13 +1,81 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        notificationButton = findViewById(R.id.notificationButton); //references notificationButton in xml by id
```
The xml with a button to be found by this line of code is missing.

kvangork commented on app/src/main/java/csustan/edu/cs4800/warriordiningapp/MainActivity.java:
```diff
@@ -1,13 +1,81 @@
+        notificationButton.setOnClickListener(new View.OnClickListener() {
+                                                  @Override
+                                                  public void onClick(View v) {
+                                                      //notification code
```
Nice demo code to display a local notification. In the future, this button will need to do a couple things, so they should probably be encapsulated in a controller class (layered & MVC architecture patterns):
1. If there's an existing notification request for this item, delete it (or mark it canceled...we should discuss further) and stop processing.
2. Otherwise, If the user hasn't already given notification permissions for our app, request them from the operating system to get a notification token.
3. Post to the server to create a notification request using the token obtained in step 2.

kvangork commented on app/src/main/java/csustan/edu/cs4800/warriordiningapp/RequestNotif.java:
```diff
@@ -0,0 +1,22 @@
+package com.example.notification;
```
We need to get our package names aligned. @Dan-Land can you help confirm that everything you've created for the app is using `csustan.edu.cs4800.warriordiningapp` ?

kvangork commented on app/src/main/java/csustan/edu/cs4800/warriordiningapp/RequestNotif.java:
```diff
@@ -0,0 +1,22 @@
+package com.example.notification;
+//irrelevant class please ignore
```
Sorry, didn't ignore. I like the idea of creating some skeleton/scaffolding classes now based on our Report 1 analysis model and trying to fill them out over time with properties and methods.
Something else to consider: this class' naming pattern (verb-noun) looks out-of-place to me in an object-oriented codebase, but I'm not sure about Java's conventions. Would "NotificationRequester" or "NotificationRequestController" be better?

Dan-Land committed:
David_Notification.jpg

- Changed format a bit.
It seems like it would mostly work but there are a couple of code errors unless I'm missing something.
isCancel isn't declared... this code is in a method that already exists so it needs/needed to be commented out.
I will try to test it with a UI soon. More focused on Demo 1 immediate goals.
Is there a reason to do this method over push notifications? Does doing it this way require the app to be open and "on" in order to send notifications?

Dan-Land committed:
minor changes.

Minor changes, nothing substantial

gaemers3232 committed:
Update MainActivity.java

gaemers3232 committed:
Notificationcontroller

how to use non static methods across classes

gaemers3232 committed:
Manifest to extend firebase

gaemers3232 committed:
request permission firebase token

still need to prevent auto initialization

gaemers3232 committed:
Update RequestNotif.java
